### ISSUE TITLE
support for running evals every `N` steps during training

### ISSUE THREAD

**Issue by skoshx** [willccbb/verifiers#13]
So basically right now there is output for each generation while training + loss, rewards and other metrics. While it's useful, I really mostly just wanna track rewards (+ vibe check) on the evaluation dataset. Currently there isn't (AFAIK) a way to do that easily.

I think we could easily get away with just creating an abstract class:
```python
class EvalTrainer(GRPOTrainer):
    def evaluate(
        self, eval_dataset=None, ignore_keys=None, metric_key_prefix: str = "eval"
    ):
        print("run eval on env!")

        return super().evaluate(self, eval_dataset, ignore_keys, metric_key_prefix)
```

Would love to hear thoughts on this :)

EDIT: step check not needed

**Comment by skoshx** [comment]
Yeah I saw that, but I was thinking it doesn't work with the multi-step env? Because we want evals to be able to call tools and use the same abstractions as we have in `verifiers`

Not sure if I'm missing something?

**Comment by willccbb** [comment]
Trainer eval just calls compute_loss without doing the backward pass, all the env rollout logic is inside of compute_loss, should work just fine

**Comment by skoshx** [comment]
Ahh right completely overlooked the logic in the TRL fork, but thats great!
